src/validators/check_id.rs

```diff
@@ -0,0 +1,69 @@
+use crate::validators::issues::*;
+
+pub fn validate(gtfs: &gtfs_structures::Gtfs) -> Vec<Issue> {
+    let r = gtfs.routes
+            .values()
+            .filter(|route| !has_id(*route))
+            .map(|route| Issue {
+                severity: Severity::Error,
+                issue_type: IssueType::MissingId,
+                object_id: route.id.to_owned(),
+                object_name: Some(format!("{}", route)),
+                related_object_id: None,
+                details: None,
+            });
+    let t = gtfs.trips
+                .values()
+                .filter(|trip| !has_id(*trip))
+                .map(|trip| Issue {
+                    severity: Severity::Error,
+                    issue_type: IssueType::MissingId,
+                    object_id: trip.id.to_owned(),
+                    object_name: Some(format!("{}", trip)),
+                    related_object_id: None,
+                    details: None,
+                });
+    let c = gtfs.calendar
+                .values()
+                .filter(|calendar| !has_id(*calendar))
+                .map(|calendar| Issue {
+                    severity: Severity::Error,
+                    issue_type: IssueType::MissingId,
+                    object_id: calendar.id.to_owned(),
+                    object_name: Some(format!("{}", calendar)),
+                    related_object_id: None,
+                    details: None,
+                });
+    let s = gtfs.stops
+                .values()
+                .filter(|&stop| !has_id(&**stop))
+                .map(|stop| Issue {
```

Comment:
to reduce the code duplication: maybe we can do:

```rust
fn make_missing_id_issue<T: gtfs_structures::Id + Display>(o: T) -> Issue {
            Issue {
                    severity: Severity::Error,
                    issue_type: IssueType::MissingId,
                    object_id: o.id.to_owned(),
                    object_name: Some(format!("{}", o)),
                    related_object_id: None,
                    details: None,
                }
}
```

and maybe make it more generic to take the severity + issueType ?

not tested though :stuck_out_tongue_winking_eye: